data-prepper-api/src/main/java/org/opensearch/dataprepper/model/plugin/kafka/AuthConfig.java

```diff
@@ -3,20 +3,42 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-package org.opensearch.dataprepper.plugins.kafka.configuration;
+package org.opensearch.dataprepper.model.plugin.kafka;
```

Comment:
Let's keep these out of `data-prepper-api`. We should reserve this project for the most common APIs, not specific technologies.

You can make use of extensions, or perhaps add a `kafka-common` project if you want to share with the `kafka` source/sink.